app-crypt/p11-kit/patches/p11_kit-0.23.12.patchset

```diff
@@ -241,3 +244,35 @@ index abff864..7efc30a 100644
 -- 
 2.15.0
 
+
+From 83b286e7300d0b609eec4d6dfdc6fce4381de0ae Mon Sep 17 00:00:00 2001
+From: fbrosson <fbrosson@localhost>
+Date: Tue, 29 May 2018 16:08:23 +0000
+Subject: fix incorrect printf format for pid_t.
+
+
+diff --git a/common/debug.c b/common/debug.c
+index bcf6e21..cf18090 100644
+--- a/common/debug.c
++++ b/common/debug.c
+@@ -140,7 +140,7 @@ p11_debug_message (int flag,
+ 	va_list args;
+ 
+ 	if (flag & p11_debug_current_flags) {
+-		fprintf (stderr, "(p11-kit:%d) ", getpid());
++		fprintf (stderr, "(p11-kit:%ld) ", getpid());
```

Comment:
It's still incorrect I'm afraid, as the `l` prefix is used for C's `long`, which have it's size differ by architectures, so you might bump into the warning again with different arches.

Reply:
Yeah, I was not really happy with this, and even wrote [this comment](https://github.com/haikuports/haikuports/pull/2617#discussion_r191491290), but since it was working fine on all architecures, I went forward with it.
(Better fixes are welcome ;)